src/Misc/environmentTextureTools.ts

```diff
@@ -218,6 +249,7 @@ export class EnvironmentTextureTools {
         let info: EnvironmentTextureInfo = {
             version: 1,

```

Comment:
Should we bump the version ? like 1.1 as it is not a breaking change ?

Reply:
Yeah, my thinking was we can keep it as is, as nothing has been removed from the JSON data. But after thinking about this again, I am beginning to think that actually it's not only about the json, but also about what kind of data is in these array buffers. 

Like when there is an .env files created with the changes here, using WebP, any application that still uses old BJS code without these changes will break reading it, as it will wrongly assume the array buffers contains png data, which they don't. And this will probably break in unexpected ways, basically with corrupted image data, instead of giving a meaningful error message.

So maybe we should bump that indeed to 2! What do you guys think?

Reply:
Yup sounds like a good plan to me :-)

Reply:
Did that now (bump version to 2)!

Changed the types a bit, `EnvironmentTextureInfo` is now a [discriminated union](https://basarat.gitbook.io/typescript/type-system/discriminated-unions) of the now separate v1 and v2 interfaces. So by checking the version, TS will narrow the type to the appropriate version. 

Added a normalize method, which takes any version and always returns the latest, so we don't have to care about differences between these versions in other parts of the codebase. This also removed the need for this `Required<>` type, as we now always return a `EnvironmentTextureInfoV2`, which is guaranteed to have `imageType` set. /cc @deltakosh 

Tested again locally, so I think this should be fine. But please re-review the latest changes! :)